src/aiida/cmdline/commands/cmd_code.py

```diff
@@ -240,29 +241,35 @@ def show(code):
 @options.OVERWRITE()
 @options.SORT()
 @with_dbenv()
-def export(code, output_file, overwrite, sort):
+def export(code, output_file, overwrite, sort, **kwargs):
     """Export code to a yaml file."""
 
-    import yaml
+    other_args = {'sort': sort, **kwargs}
 
-    code_data = {}
-
-    for key in code.Model.model_fields.keys():
-        value = getattr(code, key).label if key == 'computer' else getattr(code, key)
-
-        # If the attribute is not set, for example ``with_mpi`` do not export it, because the YAML won't be valid for
-        # use in ``verdi code create`` since ``None`` is not a valid value on the CLI.
-        if value is not None:
-            code_data[key] = str(value)
+    if output_file is not None:
+        fileformat = output_file.suffix[1:]
+    else:
+        fileformat = 'yml'
+        output_file = pathlib.Path(f'{code.label}@{code.computer.label}.{fileformat}')
```

Comment:
This will fail for `PortableCode` since it does not have a computer. Probably just do `f'{code.full_label}.{fileformat}`. I think `full_label` is defined for all codes?